rustls/src/crypto.rs

```diff
@@ -1,11 +1,17 @@
+use std::fmt::Debug;
+
 use crate::rand::GetRandomFailed;
 use crate::server::ProducesTickets;
+use crate::{Error, NamedGroup};
 
 /// *ring* based CryptoProvider.
 pub mod ring;
 
 /// Pluggable crypto galore.
 pub trait CryptoProvider: Send + Sync + 'static {
+    /// KeyExchange operations that are supported by the provider.
+    type KeyExchange: KeyExchange;
```

Comment:
Probably a place where the associated type name shouldn't equal the trait constraint name.